Bio/SeqIO/FastaIO.py

```diff
@@ -245,6 +303,12 @@ def FastaTwoLineIterator(source, alphabet=single_letter_alphabet):
         )
 
 
+def FastaNcbiIterator(source, alphabet=single_letter_alphabet):
+    for title, sequence in SimpleFastaParser(source):
+        id, name, xrefs = fasta_title_parser_auto(title)
+        yield SeqRecord(Seq(sequence, alphabet), id, name, name, dbxrefs=xrefs)
```

Comment:
The SeqRecord ``.dbxrefs`` is expected to be a list of strings of the format ``"prefix:value"``, not a dictionary.

So given:

```
>emb|CAA12345.6||gi|78 fake protein seq @#$%^[]
MAGWSCLVTGGGGFLGQRIICLLVEEKDLQEIRVLDKVFRP
GACQGTSVVIHTAS
```

we might expect ``["EMBL:CAA12345.6", ...]``

Reply:
Does this mean to ignore the `name` for most dbs (and the `chain` for PDB), only keeping `id`/`accession code`?

Reply:
this also makes complex cross-references (e.g. patent id, which needs 3 parameters) impossible

Reply:
We can standardise naming and representations of cross-references from different databases in BioPython (this also nicely solves the problem of inconsistent naming of databases in different file formats). This allows both string representation (saving space) and representation of complex cross-references, while avoiding ambiguity.

Reply:
for example, refseq is called `REFSEQ` in `.gb` files and `ref` in NCBI's FASTA. Making a standardised name e.g. (`ref`) in BioPython and writing conversion rules for each format with cross-reference support would allow reliable and lossless conversion between different formats.

Reply:
and we may use strings like `pat:US:RE33188:1` to represent cross-references with multiple params

Reply:
See https://github.com/biopython/biopython/issues/2708#issuecomment-585910340 for discussion on this